ui/src/main.rs

```diff
@@ -72,10 +79,45 @@ fn main() {
     let logger = StatisticLogger::new(file_logger);
     chain.link_around(logger);
 
+    if env::var("CORS").is_ok() {
+        debug!("using CORS");
+        chain.link_around(CorsMiddleware);
+        chain.link_after(CorsMiddleware);
+    }
+
     info!("Starting the server on {}:{}", address, port);
     Iron::new(chain).http((&*address, port)).expect("Unable to start server");
 }
 
+struct CorsMiddleware;
+
+impl AroundMiddleware for CorsMiddleware {
+    fn around(self, handler: Box<Handler>) -> Box<Handler> {
+        Box::new(move | req: &mut Request | {
+            match req.method {
+                Method::Options => Ok(Response::with((status::Ok))),
+                _ => handler.handle(req)
+            }
+        })
+    }
+}
+
+impl AfterMiddleware for CorsMiddleware {
```

Comment:
I figured that the easiest way of adding the headers was an "after" middleware. Later I added the "around" middleware for intercepting options requests. But if you prefer having only the more powerful "around" middleware, I can change this. It makes more sense if we'll detect the preflight request. And I think that you are correct that only alloworigin is required on non-preflight requests. I'll do some testing and get back to you.